card/mod/rules/set/type/set.rb

```diff
@@ -80,3 +80,14 @@ def related_sets with_self=false
     left(new: {}).related_sets with_self
   end
 end
+
+def left_type_for_nest_editor_set_selection
+  return unless  Card.fetch_id(card.name.tag).in?([StructureID, DefaultID])
+
+  set_pattern_id = Card.fetch_id card.name.left.tag
```

Comment:
This is the type/set set.  The set_pattern is the tag, right?  This looks like it's written for a rule card.

Reply:
in any case, we should probably be using methods on the card like `def set_pattern_id` or `def set_pattern_code` so that we don't have to think so hard.